app/models/rubygem.rb

```diff
@@ -177,7 +177,9 @@ def to_param
   end
 
   def pushable?
-    new_record? || versions.indexed.count.zero?
+    return true if new_record?
+    return false if versions.indexed.count.nonzero?
+    updated_at < 100.days.ago || created_at > 30.days.ago
```

Comment:
What about to move this condition to `protected?` method or similar?

You can use simple readable condition after
```ruby
new_record? || versions.indexed.none? || !protected?
```
or similar.